projects/igniteui-angular/src/lib/select/select.component.ts

```diff
@@ -312,9 +338,53 @@ export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelec
 
     /** @hidden @internal */
     public onBlur(): void {
+        if (this.ngControl && !this.ngControl.valid) {
+             this.input.valid = IgxInputState.INVALID;
+        } else {
+            this.input.valid = IgxInputState.INITIAL;
+        }
         if (!this.collapsed) {
             this.toggleDirective.close();
         }
     }
+
+    protected onStatusChanged() {
+            if ((this.ngControl.control.touched || this.ngControl.control.dirty) &&
+            (this.ngControl.control.validator || this.ngControl.control.asyncValidator)) {
+            if (this.inputGroup.isFocused) {
+                    this.input.valid = this.ngControl.valid ? IgxInputState.VALID : IgxInputState.INVALID;
+                } else {
+                this.input.valid = this.ngControl.valid ? IgxInputState.INITIAL : IgxInputState.INVALID;
+            }
+        }
+    }
+    /**
+     * @hidden @internal
+     */
+    public ngOnInit() {
+        this.ngControl = this._injector.get(NgControl, null);
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public ngAfterViewInit() {
+        if (this.ngControl) {
+            this._statusChanges$ = this.ngControl.statusChanges.subscribe(this.onStatusChanged.bind(this));
+        }
+        this.inputGroup.isRequired = this.required;
```

Comment:
Add a comment that this is set after view init on purpose, otheriwse the group itself will override it.